Tag latest fedora container

cevich opened the pull request:
Previously when a PR was merged, another build ran for all the critical
container images, along with tagging them 'latest'.  This is not ideal,
because the content can change from the time the PR build and tested the
images until when it was merged.  There is also an anticipated future
need to access the `fedora_podman` and `prior-fedora_podman` images via a
"latest" tag.

* Update image-build tasks to only run in PRs
* Simplify `ci/make_container_images.sh` to no-longer require/use a
  magic `$PUSH_LATEST` value.
* Deduplicate all FQIN references to reuse a common prefix in `$REGPFX`
* Add a new `tag_latest_images_task` that only runs on branches, and
  simply adds a `latest` tag to all container images based on the
  (as-merged) value of `$IMG_SFX` (from IMG_SFX file)

Ref: https://github.com/containers/podman/pull/22260

cevich commented:
[Test-run with `[CI:TOOLING]` set.](https://cirrus-ci.com/task/6328541428580352)

cevich commented:
[test-run with normal VM build](https://cirrus-ci.com/task/5482212794040320) (note: typically the task would only run _after_ a PR is merged).

cevich commented:
@edsantiago PTAL when you have a moment.  This simplifies tagging 'latest' for all the tooling container images, the skopeo CI container image, and the podman CI container image (for Brent's PR).  I tested it (using a `testing` tag instead of `latest`) in two instrumented tasks, results posted above.

cevich commented:
Ref: [Normal PR-level CI run](https://cirrus-ci.com/build/6268070308610048) does **not** tag any container images.

edsantiago commented:
- 20240408t125139z-f39f38d13
- 20240328t140922z-f39f38d13 &#x21D1; (built in #342)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13              | 38-1.6          | 39-1.5          | ?               | 41-0            |
|               |                |                |                |                |                |
|*kernel*       | 6.7.9-2         | 6.8.4-100       | 6.8.4-200       | 6.8.4-200       | 6.9.0-0.rc2.20240405git8cb4a9a82b21.27 |
|               |                | 6.7.10-100&nbsp;&#x21D1; | 6.7.10-200&nbsp;&#x21D1; | 6.7.10-200&nbsp;&#x21D1; | 6.9.0-0.rc1.20240326git928a87efa423.17&nbsp;&#x21D1; |
|*grub2-common* | 2.12-2          | 2.06-116        | 2.06-118        | 2.06-118        | 2.06-119        |
|               | 2.06-13+deb12u1&nbsp;&#x21D1; |                |                |                |                |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                |                |                |                |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-1        | 1.10.3-1        | 1.10.3-3        |
|               |                |                |                |                |                |
|*buildah*      | 1.33.7+ds1-1    | 1.35.1-1        | 1.35.3-1        | 1.35.0-1        | 1.35.3-1        |
|               |                |                | 1.35.2-1&nbsp;&#x21D1; |                | 1.35.0-1&nbsp;&#x21D1; |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               |                |                |                |                |                |
|*container-selinux*| ?               | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       |
|               |                |                |                |                |                |
|*containers-common*| ?               | 1-89            | 1-99            | 1-99            | 0.58.0-8        |
|               |                |                |                |                | 0.58.0-5&nbsp;&#x21D1; |
|*criu*         | 3.17.1-3        | 3.18-1          | 3.19-2          | 3.19-2          | 3.19-4          |
|               |                |                |                |                |                |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-5        |
|               |                |                |                |                | 1.14.4-1&nbsp;&#x21D1; |
|*golang*       | 2:1.22~3        | 1.21.8-1        | 1.21.9-1        | 1.21.9-1        | 1.22.2-3        |
|               |                |                | 1.21.8-1&nbsp;&#x21D1; | 1.21.8-1&nbsp;&#x21D1; | 1.22.1-4&nbsp;&#x21D1; |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-1         | 0.7.3-1         | 0.7.3-2         |
|               |                |                |                |                |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.93-2          | 7.94-1          | 7.94-1          | 7.94-1          |
|               |                |                |                |                |                |
|*passt*        | 2024-03-26      | 2024-03-26      | 2024-03-26      | 2024-03-26      | 2024-04-05      |
|               |                |                |                |                | 2024-03-26&nbsp;&#x21D1; |
|*podman*       | 4.9.4+ds1-1     | 4.9.4-1         | 4.9.4-1         | 4.9.4-1         | 5.0.1-1         |
|               | 4.9.3+ds1-1+b1&nbsp;&#x21D1; | 4.9.3-2&nbsp;&#x21D1; |                | 4.9.3-1&nbsp;&#x21D1; | 5.0.0-1&nbsp;&#x21D1; |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-1        | 1.1.12-1        | 1.1.12-3        |
|               |                |                |                |                |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        |
|               |                |                |                | 1.14.2-1&nbsp;&#x21D1; | 1.14.2-2&nbsp;&#x21D1; |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-1         | 1.2.2-1         | 1.2.2-2         |
|               |                |                |                |                |                |
|*systemd*      | 255.4-1+b1      | 253.17-1        | 254.10-1        | 254.10-1        | 255.4-1         |
|               |                |                |                |                |                |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.34-8          | 1.35-2          | 1.35-2          | 1.35-3          |
|               |                |                |                |                |                |

edsantiago commented on .cirrus.yml:
```diff
@@ -485,13 +479,52 @@ test_build-push_task:
+    script: |
+      type -P skopeo &>/dev/null || \
+          dnf install -y skopeo
+      echo "$REG_PASSWORD" | \
+          skopeo login --password-stdin --username "$REG_USERNAME" "$REGPFX"
```
This makes me so uncomfortable. Best I can tell (from looking at other repos), Cirrus does not echo the variable expansion. Do you know if there are any guarantees that it will never do so in future?

cevich replied:
Nvm, let's just go with a dedicated script, it's safer all around.

edsantiago replied:
That was my first thought but it started getting really complicated really fast. I've been convincing myself that I'm OK with this as it is.

edsantiago reviewed (commented):
Lots of questions, one suggestion, none blocking

cevich commented on .cirrus.yml:
```diff
@@ -485,13 +479,52 @@ test_build-push_task:
+          dnf install -y skopeo
+      echo "$REG_PASSWORD" | \
+          skopeo login --password-stdin --username "$REG_USERNAME" "$REGPFX"
+      for FQIN in $TOOLING_FQINS; do
```
oh woops, yes that's a good suggestion.

cevich commented on .cirrus.yml:
```diff
@@ -485,13 +479,52 @@ test_build-push_task:
+          $REGPFX/fedora_podman:c${IMG_SFX}
+          $REGPFX/prior-fedora_podman:c${IMG_SFX}
+    script: |
+      type -P skopeo &>/dev/null || \
+          dnf install -y skopeo
```
Woops, I missed that.  No it should be removed, thanks for noticing.

edsantiago replied:
Maybe it's a temporary need, until CI VMs get updated? I have no idea where the `ibi` image comes from. If it's needed temporarily, feel free to reinstate it with a big timestamped FIXME comment

cevich replied:
I looked at this closer and it should be adding it to the family but doesn't appear to be for some reason.  It's not worth fussing over for one package.

edsantiago commented on .cirrus.yml:
```diff
@@ -485,13 +479,52 @@ test_build-push_task:
+          skopeo login --password-stdin --username "$REG_USERNAME" "$REGPFX"
+      for FQIN in $TOOLING_FQINS; do
+          echo "##### Tagging $FQIN -> latest"
+          skopeo copy "docker://$FQIN" "docker://${FQIN%:*}:latest"
```
(It's OK to ignore this, but it shouldn't be marked resolved)

cevich replied:
Oh my bad, I thought you were primarily concerned with having two loops doing the same thing.  I removed that in the script, but did not simplify the names.  I can do that in the next push.

cevich replied:
Simplified loop items and moved them directly into the script.

cevich commented:
Force-push: Restore installation of skopeo.

[Test-run with [CI:TOOLING] magic string set.](https://cirrus-ci.com/task/4523690581295104)

cevich commented:
Force-push: Simplified container image name list, moved it into the script.

[Test-run with full CI VM build](https://cirrus-ci.com/task/6599969476444160)

edsantiago commented on ci/tag_latest.sh:
```diff
@@ -0,0 +1,39 @@
+    echo "##### Tagging $imgname -> latest"
+    # IMG_SFX is defined by CI system
+    # shellcheck disable=SC2154
+    skopeo copy "docker://$REGPFX/$imgname:c${IMG_SFX}" "docker://$REGPFX/${imgname%:*}:testing"
```
Much better, but you don't need the line noise (`%:*`) any more. (This is just a drive-by, I'll review again once the DO NOT MERGE commit is gone and CI is passing)

cevich replied:
Oh ya, good point.  I'll strip it out.  CI looks to have failed due to a flake.  I'll let it run and pass before re-pushing.

cevich commented:
@edsantiago I believe I've addressed all your feedback.  This version should be ready to go (the new task should *not* run as part of this PR, only after merge).

edsantiago commented:
- 20240409t145307z-f39f38d13
- 20240328t140922z-f39f38d13 &#x21D1; (built in #342)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13              | 38-1.6          | 39-1.5          | ?               | 41-0            |
|               |                |                |                |                |                |
|*kernel*       | 6.7.9-2         | 6.8.4-100       | 6.8.4-200       | 6.8.4-200       | 6.9.0-0.rc2.20240405git8cb4a9a82b21.27 |
|               |                | 6.7.10-100&nbsp;&#x21D1; | 6.7.10-200&nbsp;&#x21D1; | 6.7.10-200&nbsp;&#x21D1; | 6.9.0-0.rc1.20240326git928a87efa423.17&nbsp;&#x21D1; |
|*grub2-common* | 2.12-2          | 2.06-116        | 2.06-118        | 2.06-118        | 2.06-119        |
|               | 2.06-13+deb12u1&nbsp;&#x21D1; |                |                |                |                |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                |                |                |                |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-1        | 1.10.3-1        | 1.10.3-3        |
|               |                |                |                |                |                |
|*buildah*      | 1.33.7+ds1-1    | 1.35.1-1        | 1.35.3-1        | 1.35.0-1        | 1.35.3-1        |
|               |                |                | 1.35.2-1&nbsp;&#x21D1; |                | 1.35.0-1&nbsp;&#x21D1; |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               |                |                |                |                |                |
|*container-selinux*| ?               | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       |
|               |                |                |                |                |                |
|*containers-common*| ?               | 1-89            | 1-99            | 1-99            | 0.58.0-8        |
|               |                |                |                |                | 0.58.0-5&nbsp;&#x21D1; |
|*criu*         | 3.17.1-3        | 3.18-1          | 3.19-2          | 3.19-2          | 3.19-4          |
|               |                |                |                |                |                |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-5        |
|               |                |                |                |                | 1.14.4-1&nbsp;&#x21D1; |
|*golang*       | 2:1.22~3        | 1.21.8-1        | 1.21.9-1        | 1.21.9-1        | 1.22.2-3        |
|               |                |                | 1.21.8-1&nbsp;&#x21D1; | 1.21.8-1&nbsp;&#x21D1; | 1.22.1-4&nbsp;&#x21D1; |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-1         | 0.7.3-1         | 0.7.3-2         |
|               |                |                |                |                |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.93-2          | 7.94-1          | 7.94-1          | 7.94-1          |
|               |                |                |                |                |                |
|*passt*        | 2024-03-26      | 2024-03-26      | 2024-03-26      | 2024-03-26      | 2024-04-05      |
|               |                |                |                |                | 2024-03-26&nbsp;&#x21D1; |
|*podman*       | 4.9.4+ds1-1     | 4.9.4-1         | 4.9.4-1         | 4.9.4-1         | 5.0.1-1         |
|               | 4.9.3+ds1-1+b1&nbsp;&#x21D1; | 4.9.3-2&nbsp;&#x21D1; |                | 4.9.3-1&nbsp;&#x21D1; | 5.0.0-1&nbsp;&#x21D1; |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-1        | 1.1.12-1        | 1.1.12-3        |
|               |                |                |                |                |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        |
|               |                |                |                | 1.14.2-1&nbsp;&#x21D1; | 1.14.2-2&nbsp;&#x21D1; |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-1         | 1.2.2-1         | 1.2.2-2         |
|               |                |                |                |                |                |
|*systemd*      | 255.4-1+b1      | 253.17-1        | 254.10-1        | 254.10-1        | 255.4-1         |
|               |                |                |                |                |                |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.34-8          | 1.35-2          | 1.35-2          | 1.35-3          |
|               |                |                |                |                |                |

edsantiago commented:
LGTM

cevich commented:
Just to bring things full-circle: After merging, the [intended tagging task ran successfully](https://github.com/containers/automation_images/runs/23622904155) :grinning:
